src/test/java/com/faforever/api/mod/ModServiceTest.java

```diff
@@ -110,6 +115,21 @@ public void testExistingUid() throws Exception {
     instance.processUploadedMod(uploadedFile, new Player());
   }
 
+  @Test
+  public void testUploaderVaultBanned() throws Exception {
+    Path uploadedFile = prepareMod(TEST_MOD);
+
+    Player uploader = mock(Player.class);
+    when(uploader.getActiveBans()).thenReturn(Set.of(
+      new BanInfo()
+        .setLevel(BanLevel.VAULT)
+    ));
+
+    expectedException.expect(Forbidden.class);
```

Comment:
why different approach here?

Reply:
Not sure, which approach you are referring to but:

a) We can't use the matcher as it is no ApiException.
b) I don't want an ApiException, as this would not cause an 403 status code.
c) The test is still JUnit 4 codebase, so I did not bother to migrate it to JUnit5 assertThrows.